reports.py

```diff
@@ -183,6 +196,54 @@ def parser_assembly_stats(parser=argparse.ArgumentParser()):
 __commands__.append(('assembly_stats', parser_assembly_stats))
 
 
+def _get_samples_from_bam(bam):
+    with pysam.AlignmentFile(bam) as af:
+        return set(rg['SM'] for rg in af.header['RG'])
+def _get_chrs_from_bam(bam):
+    with pysam.AlignmentFile(bam) as af:
+        return list(sq['SN'] for sq in af.header['SQ'])
+
+def parser_coverage_only(parser=argparse.ArgumentParser()):
+    parser.add_argument('mapped_bams', nargs='+', help='Aligned-to-self mapped bam files.')
+    parser.add_argument('out_report', help='Output report file.')
+    parser.add_argument('--cov_thresholds',
+                        nargs='+',
+                        type=int,
+                        default=(1, 5, 20, 100),
+                        help='Genome coverage thresholds to report on. (default: %(default)s)')
+    util.cmd.common_args(parser, (('loglevel', None), ('version', None)))
+    util.cmd.attach_main(parser, coverage_only, split_args=True)
+    return parser
+def coverage_only(mapped_bams, out_report, cov_thresholds=(1, 5, 20, 100)):
+    header = ['sample','aln2self_cov_median', 'aln2self_cov_mean', 'aln2self_cov_mean_non0']
+    header += ['aln2self_cov_%dX' % thresh for thresh in cov_thresholds]
+    with open(out_report, 'wt') as outf:
+        outf.write('\t'.join(header)+'\n')
+        for bam in mapped_bams:
+            # check for index and auto-create if needed
+            with pysam.AlignmentFile(bam) as af:
+                is_indexed = af.has_index()
+            if not is_indexed:
```

Comment:
The user should know better, but should an error be emitted if the input bam is not aligned?

Reply:
I think that would fail the principle of "allow empty input and pass empty output" though. What if there are genuinely no aligned reads? Currently we'll just populate the output with zeros.